packages/libpulseaudio/module-sles-sink.c

```diff
@@ -307,39 +346,12 @@ static void thread_func(void *userdata) {
 finish:
     pa_log_debug("Thread shutting down");
 }
-	
-static SLint32 PA2SLrate(int32_t rate){
-	if (!(rate >= 8000 && rate <= 192000)) return -1;
-	switch(rate){
-		case 8000:
-			return SL_SAMPLINGRATE_8;
-		case 11025:
-			return SL_SAMPLINGRATE_11_025;
-		case 12000:
-			return SL_SAMPLINGRATE_12;
-		case 16000:
-			return SL_SAMPLINGRATE_16;
-		case 22050:
-			return SL_SAMPLINGRATE_22_05;
-		case 24000:
-			return SL_SAMPLINGRATE_24;
-		case 32000:
-			return SL_SAMPLINGRATE_32;
-		case 44100:
-			return SL_SAMPLINGRATE_44_1;
-		case 48000:
-			return SL_SAMPLINGRATE_48;
-		case 64000:
-			return SL_SAMPLINGRATE_64;
-		case 88200:
-			return SL_SAMPLINGRATE_88_2;
-		case 96000:
-			return SL_SAMPLINGRATE_96;
-		case 192000:
-			return SL_SAMPLINGRATE_192;
-		default:
-			return -1;
-        }
+
+static int getenv_int(const char * env, size_t min_len){
+    char * got_env = getenv(env);
+    int ret = 0;
+    if (got_env != NULL && strlen(got_env) >= min_len) ret = atoi(got_env); //"8000" is 4 symbols
```

Comment:
Same problems.

Reply:
Should we even make it a function when it's only gonna be used once?

Reply:
No it doesn't need to be a function for now. Only reason to make it a function is to see if the fast mixer sample rate has changed due to using blue tooth or even usb speaker etc and that would require calling a java api. Iv already started on one if i can find some time i might make it better but the termux-api stuff needs to be fixed to work with android 8 and that is non trivial afaik fornwall is working on that. So i have no idea for now if its worth doing if massive changes will need to be made.

Reply:
@twaik 
```
static int getenv_int(const char * env){
    char * got_env = getenv(env);
    int ret = 0;
    if (got_env != NULL)
        ret = atoi(got_env);
    return ret;
}
```

Reply:
@its-pointless yeah good luck finding a way to be notified of the audio device change.
Anyway I have been curious whether the AudioManager property thing is dynamic at all. You mind sharing *the apk* of your Termux API somewhere?

Reply:
There is no way to do that such it would work for you. I don't have the release keys....

Reply:
> There is no way to do that such it would work for you. I don't have the release keys....

Well you can sign it with your own key, but @tomty89 would need to uninstall termux-app, termux-api, etc and then install the apps you've built and signed (so quite inconvenient)

Reply:
@its-pointless Hmm so you haven't tested it yourself yet either? I thought you also built the main app yourself or so...

Reply:
@Grimler91 can't I just *re-install* in that case (so whatever in `com.termux` sticks around)?

Reply:
No iv got main app and api tested. If you really want to do that i will get around to it

Reply:
@its-pointless yeah I would love to know how well it works; it would be lovely if you can share the builds

Reply:
@tomty89 Not afaik, installation fails if com.termux contains something signed with another key. But you can of course save/backup all the files, uninstall, install its-pointless build and then restore the files.

Reply:
For this to work it requires uninstalling all termux stuff and installing test stuff with my key
https://github.com/its-pointless/its-pointless.github.io/ 
App & api.
If you can't be bothered its entirely understandable

Reply:
@its-pointless As expected, `PROPERTY_OUTPUT_SAMPLE_RATE` and `PROPERTY_OUTPUT_FRAMES_PER_BUFFER` remains unchanged on different audio devices. `STREAM_MUSIC` changes when I am on USB audio, while Bluetooth and internal audio seems to share the same value.

Reply:
@its-pointless Maybe you want to check out how the AudioTrack output driver of VLC gets the mix rate, coz I can see it successfully gets the mix rate on different audio devices (48000/44100/96000 for internal/Bluetooth/USB in my cases):
https://github.com/videolan/vlc/blob/5ca1ca1cb6c1e98a206df3463383d1e1ebcb41a3/modules/audio_output/audiotrack.c#L1041

Reply:
Iv updated my termux-api if you want to test. Using audiotrack like vlc

Reply:
@its-pointless hah so `STREAM_MUSIC` was actually the volume level of it. anyway I am going to continue the discussion in your pull request in termux-api.